bin/weewx/drivers/vantage.py

```diff
@@ -478,120 +518,182 @@ def genDavisLoopPackets(self, N=1):
         
         self.port.wakeup_console(self.max_tries, self.wait_before_retry)
         
-        # Request N packets:
-        self.port.send_data("LOOP %d\n" % N)
-        
-        ntries = 1
-        
-        for loop in range(N):
-            
-            if ntries > self.max_tries:
-                syslog.syslog(syslog.LOG_ERR, "vantage: Max retries (%d) exceeded." % self.max_tries)
-                raise weewx.RetriesExceeded("Max retries exceeded while getting LOOP data.")
-
-            try:
-                # Fetch a packet
-                _buffer = self.port.read(99)
-            except weewx.WeeWxIOError, e:
-                syslog.syslog(syslog.LOG_ERR, "vantage: LOOP #%d; read error. Try #%d" % (loop, ntries))
-                syslog.syslog(syslog.LOG_ERR, "   ****  %s" % e)
-                ntries += 1
-                continue
+        _readloop = True
+        while _readloop:
+            # Request N packets:
+            self.port.send_data("LOOP %d\n" % N)
 
-            if crc16(_buffer):
-                syslog.syslog(syslog.LOG_ERR,
-                              "vantage: LOOP #%d; CRC error. Try #%d" % (loop, ntries))
-                ntries += 1
-                continue
-            # ... decode it
-            loop_packet = self._unpackLoopPacket(_buffer[:95])
-            # Yield it
-            yield loop_packet
             ntries = 1
 
+            for loop in range(N):
+
+                if ntries > self.max_tries:
+                    syslog.syslog(syslog.LOG_ERR, "vantage: genDavisLoopPackets; max retries (%d) exceeded." %
+                                  self.max_tries)
+                    raise weewx.RetriesExceeded("Max retries exceeded while getting LOOP data.")
+
+                try:
+                    # Fetch a packet
+                    _buffer = self.port.read(99)
+                except weewx.WeeWxIOError, e:
+                    syslog.syslog(syslog.LOG_ERR, "vantage: genDavisLoopPackets; LOOP #%d; read error. Try #%d" %
+                                  (loop, ntries))
+                    syslog.syslog(syslog.LOG_ERR, "   ****  %s" % e)
+                    ntries += 1
+                    syslog.syslog(syslog.LOG_DEBUG, "vantage: genDavisLoopPackets; retry %s in loop %s error %s" %
+                                  (ntries, loop, e))
+                    continue
+                if len(_buffer) == 99:
+                    if crc16(_buffer):
+                        syslog.syslog(syslog.LOG_ERR, "vantage: genDavisLoopPackets; LOOP #%d; CRC error. Try #%d" %
+                                      (loop, ntries))
+                        ntries += 1
+                        continue
+                    # skip loop packet 0; packets 0 and 1 allways have the same time stamp
+                    if loop > 0:
```

Comment:
I've noticed this as well: that the first and second LOOP packet frequently have the same timestamp. But, they don't always. So, I chose to do nothing.

In any case, a low-level driver is not the place to make a decision on whether to discard data.